tests/remote_js/httpHandler.js

```diff
@@ -0,0 +1,95 @@
+const fetch = require('node-fetch');
+var config = require('config');
+const http = require("node:http");
+const https = require("node:https");
+
+
+// http(s) agents with keepAlive=true used to prevent socket from hanging
+// due to bug in node-fetch: https://github.com/node-fetch/node-fetch/issues/1735
+const httpAgent = new http.Agent({ keepAlive: true });
+const httpsAgent = new https.Agent({ keepAlive: true });
+const agentSelector = function (_parsedURL) {
```

Comment:
Despite adding a tiny delay before each API call, the socket hanging happened again tonight as I was running tests. I added http agents with `keepalive=true `as another workaround suggested in the [issue](https://github.com/node-fetch/node-fetch/issues/1735). Seems to be helping but I will keep track on this. Thinking to try out axios just so see if it would help